rct229/rulesets/ashrae9012019/section19/section19rule21.py

```diff
@@ -0,0 +1,510 @@
+                if (
+                    getattr_(hvac_p, "HVAC", "fan_system").get("air_energy_recovery")
+                    is not None
+                    and getattr_(
+                        hvac_p,
+                        "HVAC",
+                        "fan_system",
+                        "air_energy_recovery",
+                        "energy_recovery_type",
+                    )
+                    != ENERGY_RECOVERY.NONE
+                )
+                else False
+            )
+
+            serves_zones_that_have_dehumid_heat_recovery = (
+                True
+                if (
+                    hvac_p.get("cooling_system") is not None
+                    and hvac_p["cooling_system"].get("dehumidification_type")
+                    == DEHUMIDIFICATION.SERIES_HEAT_RECOVERY
+                )
+                else False
+            )
+
+        sys_type_heating_only = (
+            True
+            if any(
+                [
+                    baseline_system_type_compare(
+                        system_type, applicable_sys_type, False
+                    )
+                    for system_type in get_baseline_system_types(rmi_b).keys()
+                    for applicable_sys_type in APPLICABLE_SYS_TYPES
+                ]
+            )
+            else False
+        )
+
+        max_thermostat_heating_setpoint_schedule_p = max(
+            [
+                max(thermostat_schedule)
+                for thermostat_schedule_id in find_all(
+                    "$.buildings[*].building_segments[*].zones[*].thermostat_heating_setpoint_schedule",
+                    rmi_p,
+                )
+                for thermostat_schedule in find_one(
+                    f"$.schedules[?(@.id == {thermostat_schedule_id})].hourly_values",
+                    rmi_p,
+                )
+            ]
+        )
+        serves_zones_heated_to_60_or_higher_in_proposed = (
+            True
+            if max_thermostat_heating_setpoint_schedule_p > 60 * ureg("F")
+            else False
+        )
+
+        return {
+            "ER_not_req_for_heating_sys": ER_not_req_for_heating_sys,
+            "serves_zones_with_systems_likely_exhausting_toxic_etc": serves_zones_with_systems_likely_exhausting_toxic_etc,
+            "serves_kitchen_space": serves_kitchen_space,
+            "all_lighting_space_types_defined": all_lighting_space_types_defined,
+            "all_ventilation_space_types_defined": all_ventilation_space_types_defined,
+            "ER_modeled_in_proposed": ER_modeled_in_proposed,
+            "serves_zones_that_have_dehumid_heat_recovery": serves_zones_that_have_dehumid_heat_recovery,
+            "sys_type_heating_only": sys_type_heating_only,
+            "hvac_sys_and_assoc_zones_largest_exhaust_source": hvac_sys_and_assoc_zones_largest_exhaust_source,
+            "serves_zones_heated_to_60_or_higher_in_proposed": serves_zones_heated_to_60_or_higher_in_proposed,
+        }
```

Comment:
We have a lot flags and many twist logics in the RDS. I would suggest to:
1. Add `_p` and `_b` to the end of each flag to indicate the RMD used to derive this flag.
2. For derived flags (such as expcetion_1_applies) we can leave them as is.